wandb/sdk/launch/docker.py

```diff
@@ -125,22 +125,69 @@ def build_docker_image(project: _project_spec.Project, repository_uri, base_imag
     return image
 
 
-def _get_docker_image_uri(repository_uri, work_dir):
+def get_docker_command(image, docker_args=None, volumes=None, user_env_vars=None):
+    docker_path = "docker"
+    cmd = [docker_path, "run", "--rm"]
+
+    if docker_args:
+        for name, value in docker_args.items():
+            # Passed just the name as boolean flag
+            if isinstance(value, bool) and value:
+                if len(name) == 1:
+                    cmd += ["-" + name]
+                else:
+                    cmd += ["--" + name]
+            else:
+                # Passed name=value
+                if len(name) == 1:
+                    cmd += ["-" + name, value]
+                else:
+                    cmd += ["--" + name, value]
+
+    env_vars = {}  # TODO: get these from elsewhere?
+    if user_env_vars is not None:
+        for user_entry in user_env_vars:
+            if isinstance(user_entry, list):
+                # User has defined a new environment variable for the docker environment
+                env_vars[user_entry[0]] = user_entry[1]
+            else:
+                # User wants to copy an environment variable from system environment
+                system_var = os.environ.get(user_entry)
+                if system_var is None:
+                    raise ExecutionException(
+                        "This project expects the %s environment variables to "
+                        "be set on the machine running the project, but %s was "
+                        "not set. Please ensure all expected environment variables "
+                        "are set" % (", ".join(user_env_vars), user_entry)
+                    )
+                env_vars[user_entry] = system_var
+
+    if volumes is not None:
+        for v in volumes:
+            cmd += ["-v", v]
+
+    for key, value in env_vars.items():
+        cmd += ["-e", "{key}={value}".format(key=key, value=value)]
+    cmd += [image.tags[0]]
+    return cmd
```

Comment:
This would a good place to use shlex_quote

`from six.moves import shlex_quote`

Reply:
Added this once the command is converted to a string